src/components/eventForm/EventForm.tsx

```diff
@@ -0,0 +1,194 @@
+      'Description is required.',
+      'Invalid description. Only alphabets allowed.',
+    );
+
+    const locationError = validateInput(
+      eventData.location,
+      alphaRegex,
+      'Location is required.',
+      'Invalid location. Only alphabets allowed.',
+    );
+
+    if (titleError || attendeeLimitError || descriptionError || locationError) {
+      setInputs(currentInputs => ({
+        ...currentInputs,
+        title: {
+          value: currentInputs.title.value,
+          isValid: !titleError,
+          errorMessage: titleError,
+        },
+        attendeeLimit: {
+          value: currentInputs.attendeeLimit.value,
+          isValid: !attendeeLimitError,
+          errorMessage: attendeeLimitError,
+        },
+        description: {
+          value: currentInputs.description.value,
+          isValid: !descriptionError,
+          errorMessage: descriptionError,
+        },
+        location: {
+          value: currentInputs.location.value,
+          isValid: !locationError,
+          errorMessage: locationError,
+        },
+      }));
+      return;
+    }
+
+    setEventsArr(prevEvents => {
+      const updatedEvents = [...prevEvents, eventData];
+      storeData(updatedEvents, `${storageKeys.eventData}-${loginId}`);
+      return updatedEvents;
+    });
+
+    setInputs(emptyEventForm);
+    setDate(new Date());
+  };
+
+  return (
+    <View style={styles.container}>
+      <ScrollView>
+        <Input
+          label="Title"
+          invalid={!inputs.title.isValid}
+          onChangeText={(value: string) => handleInputChange('title', value)}
+          value={inputs.title.value}
+          textInputConfig={{
+            keyboardType: 'default',
+          }}
+          required
+          errorMessage={inputs.title.errorMessage}
+        />
+        <DateAndTimePicker
+          label="Select Date"
+          invalid={false}
+          date={date}
+          setDate={setDate}
+          dateOrTime="date"
+          required
+        />
+        <DateAndTimePicker
+          label="Select Time"
+          invalid={false}
+          date={date}
+          setDate={setDate}
+          dateOrTime="time"
+          required
+        />
+        <Input
+          label="Description"
+          invalid={!inputs.description.isValid}
+          onChangeText={(value: string) =>
+            handleInputChange('description', value)
+          }
+          value={inputs.description.value}
+          textInputConfig={{
+            keyboardType: 'numeric',
+            multiline: true,
+          }}
+          required
+          errorMessage={inputs.description.errorMessage}
+        />
+
+        <Input
+          label="Attendee Limit"
+          invalid={!inputs.attendeeLimit.isValid}
+          onChangeText={(value: string) =>
+            handleInputChange('attendeeLimit', value)
+          }
+          value={inputs.attendeeLimit.value}
+          textInputConfig={{
+            keyboardType: 'default',
```

Comment:
why are you passing this key everytime in custom component it should set to default